pkg/source/kafka/kafka_source.go

```diff
@@ -0,0 +1,291 @@
+//
+// Copyright (c) 2020-present Snowplow Analytics Ltd. All rights reserved.
+//
+// This program is licensed to you under the Snowplow Community License Version 1.0,
+// and you may not use this file except in compliance with the Snowplow Community License Version 1.0.
+// You may obtain a copy of the Snowplow Community License Version 1.0 at https://docs.snowplow.io/community-license-1.0
+
+package kafkasource
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/Shopify/sarama"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+	"github.com/snowplow/snowbridge/config"
+	"github.com/snowplow/snowbridge/pkg/common"
+	"github.com/snowplow/snowbridge/pkg/models"
+	"github.com/snowplow/snowbridge/pkg/source/sourceiface"
+	"github.com/twinj/uuid"
+)
+
+// Configuration configures the source for records
+type Configuration struct {
+	Brokers        string `hcl:"brokers" env:"SOURCE_KAFKA_BROKERS"`
+	TopicName      string `hcl:"topic_name" env:"SOURCE_KAFKA_TOPIC_NAME"`
+	ConsumerName   string `hcl:"consumer_name" env:"SOURCE_KAFKA_CONSUMER_NAME"`
+	OffsetsInitial int64  `hcl:"offsets_initial" env:"SOURCE_KAFKA_OFFSETS_INITIAL"`
+
+	ConcurrentWrites int    `hcl:"concurrent_writes,optional" env:"SOURCE_CONCURRENT_WRITES"`
+	Assignor         string `hcl:"assignor,optional" env:"SOURCE_KAFKA_ASSIGNOR"`
+	TargetVersion    string `hcl:"target_version,optional" env:"SOURCE_KAFKA_SOURCE_VERSION"`
+	EnableSASL       bool   `hcl:"enable_sasl,optional" env:"SOURCE_KAFKA_ENABLE_SASL"`
+	SASLUsername     string `hcl:"sasl_username,optional" env:"SOURCE_KAFKA_SASL_USERNAME" `
+	SASLPassword     string `hcl:"sasl_password,optional" env:"SOURCE_KAFKA_SASL_PASSWORD"`
+	SASLAlgorithm    string `hcl:"sasl_algorithm,optional" env:"SOURCE_KAFKA_SASL_ALGORITHM"`
+	CertFile         string `hcl:"cert_file,optional" env:"SOURCE_KAFKA_TLS_CERT_FILE"`
+	KeyFile          string `hcl:"key_file,optional" env:"SOURCE_KAFKA_TLS_KEY_FILE"`
+	CaFile           string `hcl:"ca_file,optional" env:"SOURCE_KAFKA_TLS_CA_FILE"`
+	SkipVerifyTLS    bool   `hcl:"skip_verify_tls,optional" env:"SOURCE_KAFKA_TLS_SKIP_VERIFY_TLS"`
+}
+
+// kafkaSource holds a new client for reading messages from Apache Kafka
+type kafkaSource struct {
+	config           *sarama.Config
+	concurrentWrites int
+	topic            string
+	brokers          string
+	consumerName     string
+	log              *log.Entry
+	cancel           context.CancelFunc
+
+	client sarama.ConsumerGroup
+}
+
+// consumer represents a Sarama consumer group consumer
+type consumer struct {
+	concurrentWrites int
+	throttle         chan struct{}
+	source           *sourceiface.SourceFunctions
+	log              *log.Entry
+}
+
+// Setup is run at the beginning of a new session, before ConsumeClaim
+func (consumer *consumer) Setup(sarama.ConsumerGroupSession) error {
+	consumer.log.Debugf("New session started")
```

Comment:
To me this looks a bit strange to have a log line that claims to do something, but the log line is in a function which is not doing that thing.

If you get what I mean.

More so for the `Cleanup` function.  But it doesn't bother me too much, as it's just a debug line after all.